feathr_project/feathr/job_utils.py

```diff
@@ -19,11 +19,23 @@ def get_result_df(client: FeathrClient) -> pd.DataFrame:
         # helper function for only parquet and avro
         if format.lower()=="parquet":
             files =  glob.glob(os.path.join(tmp_dir.name, '*.parquet'))
+            from pyarrow.parquet import ParquetDataset
             ds = ParquetDataset(files)
-            vertical_concat_df = ds.read().to_pandas()
+            result_df = ds.read().to_pandas()
+        if format.lower()=="delta":
+            from deltalake import DeltaTable
+            delta = DeltaTable(tmp_dir.name)
+            if not client.spark_runtime == 'azure_synapse':
+                # don't detect for synapse result with Delta as there's a problem with underlying system
+                # Issues are trached here: https://github.com/delta-io/delta-rs/issues/582
+                result_df = delta.to_pyarrow_table().to_pandas()
+            else:
+                print("Please use Azure Synapse to read the result in the Azure Synapse cluster. Reading local results is not supported for Azure Synapse. Emtpy DataFrame is returned.")
+                result_df = pd.DataFrame()
     else:
+        import pandavro as pdx
```

Comment:
nit: Move to the top

Reply:
Actually I put it here so that different formats can just import the corresponding package (just in case users don't have those installed, it won't break the other code)